doc/specs/#182 - Support for installation of portable standalone apps.md

```diff
@@ -0,0 +1,196 @@
+---
+author: Demitrius Nelon @denelon
+created on: 2022-03-09
+last updated: 2022-03-09
+issue id: 182
+---
+
+# Portable/Standalone applications
+
+"For [#182](https://github.com/microsoft/winget-cli/issues/182)"
+
+## Abstract
+
+Several packages do not have an installer. They are simply a binary executable. This specification describes how these types of programs will be treated as "packages" by the Windows Package Manager.
+
+## Inspiration
+
+Just because an installer doesn't exist for a particular package doesn't mean you shouldn't be able to install it with the Windows Package Manager.
+
+## Solution Design
+
+The installer type “portable” will be added to the enumerated list of installer types to support these types of packages.
+
+When one of these packages is encountered, there are several aspects to “installing” these packages and supporting the “upgrade” behavior. Notable concerns include the location of the package, creating or updating an entry in Windows Apps & Features.
+
+### Command Line Arguments
+
+Users should be able to specify the location for where the program is “installed” on their machine with the “--location” argument.
+
+Some packages like GitLab Runner have a file name including extra metadata (gitlab-runner-windows-386.exe). The installation instructions suggest renaming the file after it has been downloaded. In that vein, portable / standalone executables should have their “command” value specified so the Windows Package Manager can perform a default rename behavior. In addition, a “--rename” argument should be added so the user can choose a value they prefer. To have the upgrade scenario honor the custom name, this information should be recorded in the installed packages data store.
+
+Some portable applications generate or consume other files. An additional argument "--purge" will be added for the uninstall scenario to remove all files and subsequently the directory if the user wishes. The corollary argument "--preserve" will be used to preserve files if the default setting has been modified.
+
+### Settings
+
+Users should have settings to be able to specify a new default location other than the system default for the Windows Package Manager.
+
+The default path for installing these packages is "%LOCALAPPDATA%/WinGet_Packages/User" for user based installs. The default path for installing these packages is "%LOCALAPPDATA%/WinGet_Packages/Machine"for machine wide installs.
+
+The corresponding settings are "PortablePackageUserRoot" and "PortablePackageUserRoot".
+
+>Note: The "packageIdentifier" will be used to generate subdirectories for these binaries to be installed to.
+
+Using Gitlab.gitlab runner as an example would result in the .exe being placed in "%LOCALAPPDATA%/WinGet_Packages/User/Gitlab.gitlab-runner/"
```

Comment:
Do we want to specify how a conflict between identifiers from different sources would be handled?  My first thought is that by default we wouldn't include anything about the source, but if a conflict was detected, we would prepend the source name to the second (and third, etc.) directories.

Reply:
I think it's an edge case, but what if a user changes/removes that source? We thought about including the source in one iteration internally but decided against it.